### ISSUE TITLE
The argument  --fallback-style={...} for clangd is not working with coc-clangd

### ISSUE THREAD

**Issue by matevskial** [clangd/clangd#362]
# Problem

I want to set up a fallback formatter for clangd with the concrete argument `--fallback-style={--fallback-style="{BasedOnStyle: LLVM, IndentWidth: 4}"}` in file `coc-settings.json` so the formatted code has indentation width of 4, but then the coc command in nvim `:Format` still indents the code with indetntation width of 2.

This is the `coc-settings.json` config file:

```json
{
  "clangd.path":"clangd-9",
  "clangd.arguments":["--fallback-style=\"{BasedOnStyle: LLVM, IndentWidth: 4}\""]
}
```

# Expected behaviour

The coc command in nvim `:Format` should format the code with indent width 4

# Current behaviour

The formatted code is with indent width 2

![Screenshot_20200503_133847](https://user-images.githubusercontent.com/44746117/80913235-6d0b3e00-8d43-11ea-8d26-a6ac8573af10.png)

**Comment by sam-mccall** [comment]
We'll almost certainly address this when we add config files, allowing something like:

```
Format:
  FallbackStyle:
    BasedOnStyle: LLVM
    IndentWidth: 4
```

to be placed in a dotfile. It's unlikely we'll fix this in the command-line flag, as we'll likely deprecate that in favor of user config files.

**Comment by 1m188** [comment]
when write some personal small files like algorithm practice and so on, it is cumbersome to configure .clang-format every time, may be it's greater to support both config file and command-line flag?

**Comment by thebeastxdxd** [comment]
i agree i think you should still support the command line flags, config files (with 1 line) in every project is very cumbersome.

**Comment by arminveres** [comment]
Did something change here or is there a solution to this? I'm using native lsp of nvim and the issue seemingly still persists.

**Comment by cursoras** [comment]
At the moment, is there any way to change the (default) format settings, in order to be applied when the .clang-format file is missing?

**Comment by GZGavinZhao** [comment]
My current solution is to add a file at `$HOME/.clang-format`. `clangd`/`clang-format` will traverse upwards the filesystem to find the closest `.clang-format` file, so if you don't have a `.clang-format` file all the way from your current directory to your `$HOME` directory, the `$HOME/.clang-format` file will be picked up and used by `clangd`/`clang-format`.

For example:
```yaml
# ~/.clang-format
---
BasedOnStyle: Microsoft
IndentWidth: 4
---
Language: Cpp
# Force pointers to the type for C++.
DerivePointerAlignment: false
PointerAlignment: Left
```
And my file becomes:
```cpp
#include <iostream>

using namespace std;

int N, D;

int main()
{
    int* a;
    cin >> N >> D;
}
```

For Linux/Windows users, you can put the file at `/.clang-format` or `C:\.clang-format` (with `sudo`). I'm on Mac and it has this whatever system integrity thing that I don't want to break, so I just put it at my home directory.

**Comment by tae-soo-kim** [comment]
Why not add the `--style` option in `clang-format` to `clangd`?

**Comment by FalcoGer** [comment]
It would be nice if there could be a command line flag for clangd to load the format configuration file from a specified path instead of putting in a string in the command line. All my projects have the same format, and putting a format file into every one, even little test projects that live for less than a day is cumbersome. I would much rather like a fall back format file, such as the `$HOME/.config/clangd/config.yaml` file.

**Comment by HighCommander4** [comment]
> It would be nice if there could be a command line flag for clangd to load the format configuration file from a specified path instead of putting in a string in the command line.

We have https://github.com/clangd/clangd/issues/805 open for this.
